src/hooks/useAddLiquidityHandler.tsx

```diff
@@ -174,27 +175,118 @@ export const useAddLiquidityHandler = () => {
+          POSITION_MANAGER_ADDRESS || ''
+        )
+
+        let tokenBContract = new ethers.Contract(
+          tokenBSelected.address,
+          IERC20.abi,
+          accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+        )
+
+        let approvalB = await validateAllowance(
+          accountInfo.address,
+          tokenBContract,
+          BigNumber.from(parseFloat(tokenBInputVal)),
+          POSITION_MANAGER_ADDRESS || ''
+        )
+
+        let loading = notifyLoading('Waiting for transaction to complete')
+        try {
+          if (approvalA && approvalB) {
+            const DepositReservesParams = {
+              cfmm: process.env.NEXT_PUBLIC_CFMM_ADDRESS,
+              amountsDesired: [BigNumber.from(parseFloat(tokenAInputVal)), BigNumber.from(parseFloat(tokenBInputVal))],
+              amountsMin: [0, 0],
+              to: accountInfo.address,
+              protocol: 1,
+              deadline: ethers.constants.MaxUint256,
+            }
+            let tx = await posManager.depositReserves(DepositReservesParams, {
+              gasLimit: process.env.NEXT_PUBLIC_GAS_LIMIT,
+            })
+            let res = await tx.wait()
+            const { args } = res.events[0]
+            let message = 'Add Liquidity Success: ' + args.pool + args.reservesLen.toNumber() + args.shares.toNumber()
+            notifyDismiss(loading)
+            notifySuccess(message)
+          } else {
+            notifyDismiss('Waiting for transaction to complete')
+          }
+        } catch (e: any) {
+          notifyDismiss(loading)
+          if (e?.code === 'ACTION_REJECTED') {
+            notifyError('User rejected the transaction')
+          } else {
+            notifyError('An error occurred while adding LP Token Liquidity. Please try again')
+          }
+        }
+      }
+    }
+  }
+
+  const addLpLiquidity = async () => {
+    if (posManager && accountInfo?.address && provider) {
+      let tokenContract = new ethers.Contract(
+        tokenASelected.address,
+        IERC20.abi,
+        accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+      )
+
+      let approval = await validateAllowance(
+        accountInfo.address,
+        tokenContract,
+        BigNumber.from(parseFloat(tokenAInputVal)),
+        POSITION_MANAGER_ADDRESS || ''
+      )
+      let loading = notifyLoading('Waiting for transaction to complete')
       try {
-        const DepositReservesParams = {
-          cfmm: process.env.NEXT_PUBLIC_CFMM_ADDRESS,
-          amountsDesired: [BigNumber.from(parseFloat(tokenAInputVal)), BigNumber.from(parseFloat(tokenBInputVal))],
-          amountsMin: [0, 0],
-          to: accountInfo.address,
-          protocol: 1,
-          deadline: ethers.constants.MaxUint256,
+        if (approval) {
+          const DepositNoPullParams = {
+            cfmm: tokenASelected.address,
```

Comment:
tokenASelected is not a uniPair, it's just a reserve token, use a uniPair address instead of a reserve token address

Reply:
Done